examples/blend.cpp

```diff
@@ -119,33 +206,70 @@ class Blend
             memset(&m_dest->crop, 0, sizeof(VideoRect));
             m_scaler->process(frame, m_dest);
 
-            PERF_START(blend);
-            //blend it
-            for (uint32_t i = 0; i < m_blendSurfaces.size(); i++) {
-                m_blendBumpBoxes[i]->getPos(m_dest->crop.x, m_dest->crop.y, m_dest->crop.width, m_dest->crop.height);
-                //printf("(%d, %d, %d, %d)\r\n", m_dest->crop.x, m_dest->crop.y, m_dest->crop.width, m_dest->crop.height);
+            if (m_bBlend) {
+                PERF_START(blend);
+                //blend it
+                for (uint32_t i = 0; i < m_blendSurfaces.size(); i++) {
+                    m_blendBumpBoxes[i]->getPos(m_dest->crop.x, m_dest->crop.y, m_dest->crop.width, m_dest->crop.height);
+                    //printf("(%d, %d, %d, %d)\r\n", m_dest->crop.x, m_dest->crop.y, m_dest->crop.width, m_dest->crop.height);
+
+                    SharedPtr<VideoFrame>& src = m_blendSurfaces[i];
+                    m_blender->process(src, m_dest);
+                }
+                PERF_STOP(blend);
+            }
+
+            if (m_bOsd) {
+                PERF_START(osd);
+                //osd
+                for (uint32_t i = 0; i < m_osdSurfaces.size(); i++) {
```

Comment:
duplicate code. could we add one function like
DrawSrufaces(m_osdSurfaces, m_osdBumpBoxes, m_osd)